lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -670,9 +670,10 @@ namespace Js
             // Guard the closing and deleting of DebugContext as in meantime PDM might
             // call OnBreakFlagChange
             AutoCriticalSection autoDebugContextCloseCS(&debugContextCloseCS);
-            this->debugContext->Close();
-            HeapDelete(this->debugContext);
+            DebugContext* tempDebugContext = this->debugContext;
```

Comment:
I think that is not necessary as context switch between these 2 instructions won't cause an issue as InternalClose is non-entrant and no where else scriptContext->debugContext is changed.